File: crates/harn-hostlib/tests/fixtures/ast/csharp/source.cs
namespace Fixture
{
    public class Greeter
    {
        public string Name { get; set; }

        public string Greet()
        {
            return "hello " + Name;
        }
    }

    public interface ISpeaker
    {
        string Speak();
    }

    public enum Color
    {
        Red,
        Green,
        Blue
    }
}

File: crates/harn-hostlib/tests/fixtures/ast/cpp/source.cpp
namespace fixture {

class Greeter {
public:
    Greeter(const char *name) : name_(name) {}
    std::string greet() const { return std::string("hello ") + name_; }

private:
    const char *name_;
};

struct Point {
    int x;
    int y;
};

enum Color { RED, GREEN, BLUE };

int add(int a, int b) {
    return a + b;
}

}
